components/editor/utils/OverlayForm.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react'
+import ReactDOM from 'react-dom'
+import { css } from 'glamor'
+
+import {
+  Overlay,
+  OverlayToolbar,
+  OverlayToolbarClose,
+  OverlayBody,
+  mediaQueries
+} from '@project-r/styleguide'
+
+const previewWidth = 290
+
+const styles = {
+  editButton: css({
+    position: 'absolute',
+    left: -40,
+    top: 0,
+    zIndex: 1,
+    fontSize: 24,
+    ':hover': {
+      cursor: 'pointer'
+    }
+  }),
+  preview: css({
+    [mediaQueries.mUp]: {
+      float: 'left',
+      width: previewWidth
+    }
+  }),
+  edit: css({
+    [mediaQueries.mUp]: {
+      float: 'left',
+      width: `calc(100% - ${previewWidth}px)`,
+      paddingLeft: 20
+    }
+  })
+}
+
+class OverlayForm extends Component {
+  constructor (...args) {
+    super(...args)
+    this.rootDiv = document.createElement('div')
+    document.body.appendChild(this.rootDiv)
+  }
+  render () {
+    const { onClose, preview, extra, children } = this.props
+
+    return ReactDOM.createPortal(
+      <Overlay onClose={onClose} mUpStyle={{maxWidth: '80vw', marginTop: '5vh'}}>
+        <OverlayToolbar>
+          <OverlayToolbarClose onClick={onClose} />
+        </OverlayToolbar>
+
+        <OverlayBody>
+          <div {...styles.preview}>
+            {preview}
+            <br />
+            {extra}
+          </div>
+          <div {...styles.edit}>
+            {children}
+          </div>
+          <br style={{clear: 'both'}} />
+        </OverlayBody>
+      </Overlay>,
+      this.rootDiv
+    )
+  }
+}
+
+export default OverlayForm
```

Comment:
I think this would be a situation where prop-types could help with eventual reuses of the component. They would add a minimal layer of documentation and property validation.